src/axom/slic/streams/GenericOutputStream.hpp

```diff
@@ -49,6 +49,12 @@ class GenericOutputStream : public LogStream
    *   - "cerr" makes std::cerr the output stream
    *   - Any other input will construct a std::ofstream associated with input
    * \param [in] stream the string to control type of stream created
+   *
+   * \note The constructed std::ofstream will open and associate a file with
+   *       the stream the first time GenericOutputStream is flushed with at
+   *       least one buffered message. Use this constructor
+   *       to avoid empty file creation if GenericOutputStream is not appended
+   *       to.
```

Comment:
This `note` (and the one at new line 82) is hard to understand.  Please rephrase.  Consider dropping the first sentence and changing the second to "Use this constructor to avoid creating an empty file if this GenericOutputStream never gets a message."

Reply:
Fixed!